### PR TITLE
Change preventing accidental new mail discard

### PR DESCRIPTION
**Pull request by jcontno**
Hi,

I added a confirm dialog if accidently pressed discard menu entry in the message compose string. The dialog is default on and can be disabled in global settings > interaction > notifications

Cheers
Joerg

P.S: Being new on github. Once you accepted the pull (and if I need space), can I delete my fork ?

### CODE REVIEW

**Comment by cketti** [comment]
There are some code style issues. Please read the [code style](https://github.com/k9mail/k-9/wiki/CodeStyle) wiki page.

Please only add/change English strings. We use [Transifex](https://www.transifex.com/projects/p/k9mail/) for translations (and there's no two-way sync).

**Comment by jcontno** [comment]
I just saw the comments and removing the values-de string. Now on the branch it should be in good shape. Or see https://github.com/jcontno/k-9/commit/57c3d0a4249f9577f984897ed43b62df16a78be8 

Am sorry I don't know what to do at transifex, maybe someone else can add the line or give me a specific pointer.

Good now ?

**Comment by cketti** [comment]
You don't use the framework to display the dialog. This won't recreate the dialog when you rotate the device. See `MessageCompose.onCreateDialog()`

When adding a global setting one also has to add it to `com.fsck.k9.preferences.GlobalSettings` so import/export will work.

In my opinion a simple "Discard message" works better than "Discard at Message Compose Menu".

File: k9mail/src/main/java/com/fsck/k9/K9.java [hunk 1]
```diff
@@ -168,6 +168,7 @@
     private static boolean mAnimations = true;
 
     private static boolean mConfirmDelete = false;
+    private static boolean mConfirmMenuDiscard = true;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/K9.java]: The field name doesn't really capture what this option does. I think `mConfirmDiscardMessage` would be a better choice.

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 2]
```diff
@@ -1939,7 +1939,28 @@ private void showOrHideQuotedText(QuotedTextMode mode) {
             }
         }
     }
-
+    
+    private void askBeforeDiscard(){
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: This method still contains quite a few code style issues. Missing spaces, wrong indentation, lines that are too long.

Importing the settings to Android Studio, selecting all text of the method, then using "reformat code" should do the right thing. See https://github.com/k9mail/k-9/wiki/CodeStyle

File: k9mail/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -101,7 +101,7 @@ Um Fehler zu melden, neue Funktionen vorzuschlagen oder Fragen zu stellen, besuc
   <string name="single_message_options_action">Senden…</string>
   <string name="refile_action">Umsortieren…</string>
   <string name="done_action">Fertig</string>
-  <string name="discard_action">Verwerfen</string>
+  <string name="discard_action">Verwerfen </string>
```
Comment (cketti) [inline k9mail/src/main/res/values-de/strings.xml]: The pull request still changes `values-de/strings.xml` :(

**Comment by cketti** [comment]
I know it's a lot of work for such a simple feature. But don't let this discourage you. Once you're done you will be one of the 100+ people who contributed to K-9 Mail. Bragging about that is strongly encouraged :)

**Comment by cketti** [comment]
I've done the changes myself.

https://github.com/k9mail/k-9/commit/10cb346c198b481c01958ffa3fca6de4b926b78c

**Pull request closed without merging**